paseos/attitude/attitude_model.py

```diff
@@ -0,0 +1,300 @@
+                    earth_rotation_vector_in_rpy, euler_angles
+                )
+                # Accumulate torque due to gravity gradients
+                T += compute_gravity_gradient_torque(
+                    self._actor.central_body.planet,
+                    nadir_vector_in_body,
+                    earth_rotation_vector_in_body,
+                    self._actor.body_moment_of_inertia,
+                    np.linalg.norm(position),
+                )
+            if "magnetic" in self._actor.attitude_disturbances:
+                time = self._actor.local_time
+                T += compute_magnetic_torque(
+                    m_earth=self._actor.central_body.magnetic_dipole_moment(time),
+                    m_sat=self._actor_residual_magnetic_field,
+                    position=self._actor.get_position(time),
+                    velocity=self._actor.get_position_velocity(time)[1],
+                    attitude=self._actor_attitude_in_rad,
+                )
+        return T
+
+    def _calculate_angular_acceleration(self, current_temperature_K):
+        """Calculate the spacecraft angular acceleration (external disturbance torques and gyroscopic accelerations).
+        Args:
+            current_temperature_K (float): current temperature in Kelvin.
+        """
+
+        # TODO in the future control torques could be added
+        # Euler's equation for rigid body rotation: a = I^(-1) (T - w x (Iw))
+        # with: a = angular acceleration, body_moment_of_inertia = inertia matrix, T = torque vector, w = angular velocity
+        self._actor_angular_acceleration = np.linalg.inv(self._actor.body_moment_of_inertia) @ (
+            self.compute_disturbance_torque(
+                position=np.array(self._actor.get_position(self._actor.local_time)),
+                velocity=np.array(self._actor.get_position_velocity(self._actor.local_time)[1]),
+                euler_angles=self._actor_attitude_in_rad,
+                current_temperature_K=current_temperature_K,
+            )
+            - np.cross(
+                self._actor_angular_velocity,
+                self._actor.body_moment_of_inertia @ self._actor_angular_velocity,
+            )
+        )
+
+    def _body_rotation(self, dt, current_temperature_K):
+        """Calculates the rotation vector around which the spacecraft body rotates
+        based on angular acceleration, velocity, and current temperature.
+
+        Args:
+            dt (float): time to advance.
+            current_temperature_K (float): current temperature in Kelvin.
+
+        Returns: rotation vector of spacecraft body expressed in the RPY frame.
+        """
+        # Calculate angular acceleration
+        self._calculate_angular_acceleration(current_temperature_K)
+
+        # Add angular velocity
+        self._actor_angular_velocity += self._actor_angular_acceleration * dt
+
+        # Body rotation vector:
+        body_rotation = self._actor_angular_velocity * dt
```

Comment:
this is a numerical model that will not be stable for large dt. We should impose constraints on the value the user can pick for dt when these models are activated

Reply:
Typically you need to set it based some stability criteria. It depends on the forces / acceleration but also how smooth / non-smooth the acceleration is.

I would suggest to set it based on the current angular velocity & acceleration amplitude (e.g. users defines some value `dt`. Then, we have some condition like

(sensible values have to be tested!)
`dt = np.minimum(dt,some_constant * (norm(angular_velocity) + norm(angular_acceleration))`

Not sure if this needs to happen outside the model to update temperature etc.? We might be able to get away with sub-timestepping this specific model? (meaning PASEOS timestep can be larger and basically, e.g. PASEOS advances 1s and this model then does e.g. 100 timesteps internally to compute rotation after 1s?) Otherwise this will severely slow down the simulation